src/index/base.h

```diff
@@ -26,14 +26,37 @@ struct IndexSummary {
  */
 class BaseIndex : public CValidationInterface
 {
+public:
+    /// Destructor interrupts sync thread if running and blocks until it exits.
+    virtual ~BaseIndex();
```

Comment:
You don't need to mark it virtual, there is a bug and it is destructor of `CValidationInterface` should marked as `virtual`